# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1246022131 14400
#      Fri Jun 26 09:15:31 2009 -0400
# Node ID 6c421f2355b5fdc0c30b60f03e2d5e48062a9485
# Parent  df1ea906c1c433875abacba4e07f420e6b92a2cc
load-path.cc (Faddpath): preserve order of prepended elements

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,12 @@
+2009-06-26  John W. Eaton  <jwe@octave.org>
+
+	* load-path.cc (Faddpath): Preserve order of prepended elements.
+
 2009-06-25  John W. Eaton  <jwe@octave.org>
 
 	* graphics.h.in (gh_manager::restore_gcbo):
 	Arg is now void, not void*.
 
 2009-06-25  Jaroslav Hajek  <highegg@gmail.com>
 
 	* unwind-prot.h (unwind_protect::fcn_elem): New class.
diff --git a/src/load-path.cc b/src/load-path.cc
--- a/src/load-path.cc
+++ b/src/load-path.cc
@@ -2019,16 +2019,19 @@ Directories added to the path must exist
       for (int i = 0; i < nargin; i++)
 	{
 	  std::string arg = args(i).string_value ();
 
 	  if (! error_state)
 	    {
 	      std::list<std::string> dir_elts = split_path (arg);
 
+	      if (! append)
+		std::reverse (dir_elts.begin (), dir_elts.end ());
+
 	      for (std::list<std::string>::const_iterator p = dir_elts.begin ();
 		   p != dir_elts.end ();
 		   p++)
 		{
 		  std::string dir = *p;
 
 		  //dir = regexprep (dir_elts{j}, "//+", "/");
 		  //dir = regexprep (dir, "/$", "");
